optimisation/ScanSelectionsTree.py

```diff
@@ -200,54 +252,100 @@
             nBinsVar0 = int((maxVar0 - minVar0) / cutVars[varNames[0]]['step'])
             nBinsVar1 = int((maxVar1 - minVar1) / cutVars[varNames[1]]['step'])
             hEstimVsCut[iPt][est] = TH2F(f'h{est}VsCut_pT{ptMin}-{ptMax}',
-                                         f';{varNames[0]};{varNames[1]};{estNames[est]}',
-                                         nBinsVar0, minVar0, maxVar0, nBinsVar1, minVar1, maxVar1)
-
+                                            f';{varNames[0]};{varNames[1]};{estNames[est]}',
+                                            nBinsVar0, minVar0, maxVar0, nBinsVar1, minVar1, maxVar1)
     startTime = time.time()
+
+    # DCA
+    if enableDCA:       
+        lDCAPrompt_bincontents, lDCAFD_bincontents = [], []
+        nTotDCAPrompt, nTotDCAFD = 0, 0
+        for iDCA, (dcaMin, dcaMax) in enumerate(zip(dcaMins, dcaMaxs)):
+            if inputCfg['DCA']['verbose']:
+                print(f'\33[1;92m DCA bincontents extraction: {dcaMin}-{dcaMax}\33[0m\r')
+            nTotDCAPrompt = float(len(dfPromptPt['dca']))
+            nTotDCAFD = float(len(dfFDPt['dca']))
+            dfDCAPrompt = dfPromptPt.query(f'{dcaMin} < dca < {dcaMax}')
+            dfDCAFD = dfFDPt.query(f'{dcaMin} < dca < {dcaMax}')
+            lDCAPrompt_bincontents.append(len(dfDCAPrompt['dca']))
+            lDCAFD_bincontents.append(len(dfDCAFD['dca']))
+            if inputCfg['DCA']['verbose']:
+                 print(f'\33[32mlDCAPrompt_bincontents: {lDCAPrompt_bincontents}\33[0m\r')
+                 print(f'\33[32mlDCAFD_bincontents: {lDCAFD_bincontents}\33[0m\r')
+            del dfDCAPrompt, dfDCAFD 
+
+        nbins = len(lDCAPrompt_bincontents)
+        binedgeslow_DCA = np.asarray(sorted(inputCfg['DCA']['DCAmin']))
+        binedgesmax_DCA = np.asarray(sorted(inputCfg['DCA']['DCAmax']))
+        cDCA = TCanvas(f'cDCA_pT{ptMin}-{ptMax}', '', 1920, 1080)
+        hDCA_Prompt = TH1F(f'hDCA_Prompt_pT{ptMin}-{ptMax}',f'DCA Prompt pT{ptMin}-{ptMax} (cm)', nbins-1, binedgeslow_DCA)
+        hDCA_FD = TH1F(f'hDCA_FD_pT{ptMin}-{ptMax}',f'DCA FD pT{ptMin}-{ptMax} (cm)', nbins-1, binedgeslow_DCA)
+        
+        if inputCfg['DCA']['verbose']:
+            print(f'\33[1;92m DCA histo filling\33[0m\r')
+        for ibin in range(nbins):
```

Comment:
Instead of looping you could use the fill_hist function of root_numpy since it is faster and it is already used in this script. You can then use the Scale(1. / histo.Integral()) function to normalise the histograms. This would also allow to remove all the lines from 261 to 279. 
I would also not put this plot here since it is not relevant for the selection scan (we can keep it if needed for the moment)